pulsar-io/kinesis/src/main/java/org/apache/pulsar/io/kinesis/KinesisRecordProcessor.java

```diff
@@ -112,4 +79,45 @@ private void checkpoint(IRecordProcessorCheckpointer checkpointer) {
         }
     }
 
+    @Override
+    public void initialize(InitializationInput initializationInput) {
+        kinesisShardId = initializationInput.shardId();
+    }
+
+    @Override
+    public void processRecords(ProcessRecordsInput processRecordsInput) {
+
+        log.info("Processing " + processRecordsInput.records().size() + " records from " + kinesisShardId);
```

Comment:
does "records" ever return null here? Would hate to throw a null pointer exception in a log statement. Do we need to do a null block?

Reply:
it should not, as the kcl v2 with enhanced fan-out is all pushed based, so it shouldn't happen